packages/foam-core/src/note-graph.ts

```diff
@@ -15,25 +15,41 @@ export interface NoteLink {
   position: Position;
 }
 
+export interface NoteLinkDefinition {
+  label: string;
+  url: string;
+  title?: string;
+  position?: Position;
+}
+
 export class Note {
   public id: ID;
   public title: string;
   public source: string;
   public path: string;
+  public end: Point;
+  public eol: string;
```

Comment:
As per our discussion with @jevakallio , we decided to store it on the note level. This might be useful when the same workspace is edited from different platforms.

Reply:
@riccardoferretti we don't *expect* them to be, but realistically they may be, so might as well do the right thing here and avoid a weird bug down the line.